### PR TITLE
Fix dropout static when axis != None

### PR DESCRIPTION
**Pull request by smallv0221**
### PR types

Bug fixes
### PR changes

APIs
### Describe

Fix dropout static when axis != None

### CODE REVIEW

File: python/paddle/nn/functional/common.py [hunk 1]
```diff
@@ -940,6 +940,7 @@ def get_attrs(prog, dropout_prob, is_test, seed):
 
             #get mask shape
             input_shape = x.shape
+            input_shape_tensor = paddle.shape(x)
```
Comment (guoshengCS) [inline python/paddle/nn/functional/common.py]: 为了减小对性能的影响，是否区分动静态图呢

Comment (smallv0221) [reply python/paddle/nn/functional/common.py]: 是说动态图的时候可以少走一个op是么

File: python/paddle/fluid/tests/unittests/test_dropout_op.py [hunk 2]
```diff
@@ -380,7 +380,11 @@ def check_static_result(self, place):
                 training=False,
                 mode='upscale_in_train')
 
-            in_np = np.random.random([40, 40]).astype("float32")
+            res13 = paddle.nn.functional.dropout(
+                x=input, p=0.7, axis=1, training=True, mode='upscale_in_train')
+
+            in_np = np.ones([40, 40]).astype("float32")
+            in_np2 = np.ones([1, 250000000]).astype("float32")
```
Comment (guoshengCS) [inline python/paddle/fluid/tests/unittests/test_dropout_op.py]: 因为这个指定axis的分支不会走dropout op，感觉加的单测不用和去dropout OP比较，能跑下来就行，这个大size的单测可能会带来其他问题

Comment (smallv0221) [reply python/paddle/fluid/tests/unittests/test_dropout_op.py]: done

**Review by guoshengCS — APPROVED** [review]
